### ISSUE TITLE
iter with longest_prefix match

### ISSUE THREAD

**Issue by Huarong** [WojciechMula/pyahocorasick#108]
now:

```ipython
In [10]: import ahocorasick

In [11]: >>> for idx, key in enumerate('he her hers she'.split()):
    ...: ...   A.add_word(key, (idx, key))
    ...:

In [12]: A.make_automaton()

In [13]: list(A.iter('he her'))
Out[13]: [(1, (0, 'he')), (4, (0, 'he')), (5, (1, 'her'))]
```

expect:

```ipython
In [13]: list(A.iter('he her', longest_prefix=True))
Out[13]: [(1, (0, 'he')), (5, (1, 'her'))]
```

**Comment by snoopyjc** [comment]
Yes, I would love this!

**Comment by snoopyjc** [comment]
Is this the same as iter_long?

**Comment by WojciechMula** [comment]
@snoopyjc Yes, that's the same. Personally I'm rather for a new method (now ``iter_long``) than an extra argument. I feel like overloading is not always the best way.

**Comment by zhouxinhit** [comment]
> Is this the same as iter_long?

I cannot find iter_long function in 1.4.0 version

**Comment by pombredanne** [comment]
This has been implemented in version 1.4.2. I am closing this now. Thank you all! :heart: 
See the doc at https://pyahocorasick.readthedocs.io/en/latest/ for details
